bigtop-packages/src/common/ambari/patch14-LOG4J.diff

```diff
@@ -0,0 +1,520 @@
++      </exclusions>
+     </dependency>
+ 
+     <dependency>
+@@ -156,6 +162,10 @@
+           <groupId>com.fasterxml.jackson.core</groupId>
+           <artifactId>jackson-databind</artifactId>
+         </exclusion>
++        <exclusion>
++          <groupId>log4j</groupId>
++          <artifactId>log4j</artifactId>
++        </exclusion>
+       </exclusions>
+     </dependency>
+     <dependency>
+diff --git a/ambari-infra/ambari-infra-solr-client/pom.xml b/ambari-infra/ambari-infra-solr-client/pom.xml
+index ff94b2a4..276e75ca 100644
+--- a/ambari-infra/ambari-infra-solr-client/pom.xml
++++ b/ambari-infra/ambari-infra-solr-client/pom.xml
+@@ -89,11 +89,17 @@
+       <groupId>org.slf4j</groupId>
+       <artifactId>slf4j-log4j12</artifactId>
+       <version>1.7.20</version>
++      <exclusions>
++        <exclusion>
++          <groupId>log4j</groupId>
++          <artifactId>log4j</artifactId>
++        </exclusion>
++      </exclusions>
+     </dependency>
+     <dependency>
+-      <groupId>log4j</groupId>
+-      <artifactId>log4j</artifactId>
+-      <version>1.2.17</version>
++      <groupId>ch.qos.reload4j</groupId>
++      <artifactId>reload4j</artifactId>
++      <version>1.2.25</version>
+       <exclusions>
+         <exclusion>
+           <groupId>com.sun.jdmk</groupId>
+@@ -190,4 +196,4 @@
+       </plugin>
+     </plugins>
+   </build>
+-</project>
+\ No newline at end of file
++</project>
+diff --git a/ambari-infra/pom.xml b/ambari-infra/pom.xml
+index 13fb6e93..6a2e49ef 100644
+--- a/ambari-infra/pom.xml
++++ b/ambari-infra/pom.xml
+@@ -324,6 +324,12 @@
+         <groupId>org.apache.zookeeper</groupId>
+         <artifactId>zookeeper</artifactId>
+         <version>${zookeeper.version}</version>
++        <exclusions>
++          <exclusion>
++            <groupId>log4j</groupId>
++            <artifactId>log4j</artifactId>
++          </exclusion>
++        </exclusions>
+       </dependency>
+       <dependency>
+         <groupId>commons-io</groupId>
+diff --git a/ambari-logsearch/ambari-logsearch-appender/pom.xml b/ambari-logsearch/ambari-logsearch-appender/pom.xml
+index fc2d03b7..fb62d3d1 100644
+--- a/ambari-logsearch/ambari-logsearch-appender/pom.xml
++++ b/ambari-logsearch/ambari-logsearch-appender/pom.xml
+@@ -62,9 +62,9 @@
+ 
+   <dependencies>
+     <dependency>
+-      <groupId>log4j</groupId>
+-      <artifactId>log4j</artifactId>
+-      <version>1.2.17</version>
++      <groupId>ch.qos.reload4j</groupId>
++      <artifactId>reload4j</artifactId>
++      <version>1.2.25</version>
+       <exclusions>
+         <exclusion>
+           <groupId>com.sun.jdmk</groupId>
+diff --git a/ambari-logsearch/ambari-logsearch-it/pom.xml b/ambari-logsearch/ambari-logsearch-it/pom.xml
+index d7a06f0e..3d94c2a3 100644
+--- a/ambari-logsearch/ambari-logsearch-it/pom.xml
++++ b/ambari-logsearch/ambari-logsearch-it/pom.xml
+@@ -96,6 +96,12 @@
+       <groupId>org.slf4j</groupId>
+       <artifactId>slf4j-log4j12</artifactId>
```

Comment:
Does it work? The  slf4j-log4j12 with excluded log4j? We should upgrade to 1.7.35 each slf4j-log4j12 https://www.slf4j.org/manual.html#swapping or use this https://mvnrepository.com/artifact/org.slf4j/slf4j-reload4j
Please research on this more so we will be sure it will work correctly

Reply:
Yes, I am able to build it, no errors and also there is no vulnerability. Could you please help me setup a running cluster to test it ?